### ISSUE TITLE
Issue with neural network output

### ISSUE THREAD

**Issue by TheoPaput** [tensorflow/tensorflow#27787]
I'm trying to create a neural network which take as input a table with the format : [[9][5]] and which got 3 outputs but as output i got a table with this format : [[5][3]] and i don'y understand why

**System information**
- tensorflow 1.13

Here my network class :

```
class DQNetwork:
    def __init__(self, state_size, action_size, learning_rate, name='DQNetwork'):
        self.state_size = state_size
        self.action_size = action_size
        self.learning_rate = learning_rate
        
        with tf.variable_scope(name):
            # We create the placeholders
            # *state_size means that we take each elements of state_size in tuple hence is like if we wrote
            # [None, 84, 84, 4]
            self.inputs_ = tf.placeholder(tf.float32, shape=[state_size[1], state_size[0]], name="inputs")
            self.actions_ = tf.placeholder(tf.float32, [None, self.action_size], name="actions_")
            
            # Remember that target_Q is the R(s,a) + ymax Qhat(s', a')
            self.target_Q = tf.placeholder(tf.float32, [None], name="target")
            
			
            self.fc = tf.layers.dense(inputs = self.inputs_,
                                      units = 50,
									  kernel_initializer=tf.contrib.layers.xavier_initializer(),
                                      activation = tf.nn.elu)
			
            
            self.output = tf.layers.dense(inputs = self.fc, 
                                        units = self.action_size,
										kernel_initializer=tf.contrib.layers.xavier_initializer(),
                                        activation=None)
            

  
            # Q is our predicted Q value.
            self.Q = tf.reduce_sum(tf.multiply(self.output, self.actions_))
            
            # The loss is the difference between our predicted Q_values and the Q_target
            # Sum(Qtarget - Q)^2
            self.loss = tf.reduce_mean(tf.square(self.target_Q - self.Q))
            
            self.optimizer = tf.train.AdamOptimizer(self.learning_rate).minimize(self.loss)
```

Here a set of input i can give to my network :

```
state = []
	state.append([0,0,0,0,0,0,0,0,0])
	state.append([0,1,0,0,0,0,0,0,0])
	state.append([0,0,0,0,1,0,0,0,0])
	state.append([0,0,1,0,0,1,0,0,0])
	state.append([0,0,0,0,0,0,0,0,0])
```

then state is a possible input

As output i expect something like : [[0.125, 0.56, 0.301]]

But i got something like :

```
[ [-0.06873338  0.12694651 -0.30928543]
 [-0.06873338  0.12694651 -0.30928543]
 [-0.06873338  0.12694651 -0.30928543]
 [-0.06873338  0.12694651 -0.30928543]
 [-0.06955577  0.12819782 -0.31083506]]
```

**Comment by TheoPaput** [comment]
I added a flatten layer trying to force the dimension of the inputs to be a 1 dimensional array like this :

```
# hidden layer
            self.hidden = tf.layers.dense(inputs = self.inputs_,
			                              units = 50,
			                              kernel_initializer=tf.contrib.layers.xavier_initializer(),
										  activation = tf.nn.elu)
										  
		    # flat the data in a one dimension array
            self.flat = tf.layers.flatten(inputs = self.hidden)
										 
			# output layer
            self.output = tf.layers.dense(inputs = self.flat,
			                              units = self.action_size, 
										  kernel_initializer=tf.contrib.layers.xavier_initializer())
```

Changed absolutely nothing .. i don't know what i missed

**Comment by jvishnuvardhan** [comment]
This is not Build/Installation or Bug/Performance issue. Please post this kind of support questions at [Stackoverflow](https://stackoverflow.com/questions/tagged/tensorflow). There is a big community to support and learn from your questions. GitHub is mainly for addressing bugs in installation and performance. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by jvishnuvardhan** [comment]
Closing as it was a support question. Thanks!
